mantidproject/mantid
# Extract detector grouping options from Indirect Data Reduction

### Description of work
This PR extracts the grouping options found in the Indirect Data Reduction interface into its own widget called `DetectorGroupingOptions`. This will make it easier to reuse this widget in the Indirect Diffraction interface.

Fixes #36763

### To test:
Follow the testing instructions here
https://github.com/mantidproject/mantid/blob/main/dev-docs/source/Testing/Indirect/DataReductionTests.rst

In particular, notice that the number of groups should correspond to the number of spectra in the reduced workspace (unless the number of detectors is not divisible by the number of groups, in which case their will be an extra group).

Try a `Custom` grouping such as `3-10,11-22,23`. Then do `Show Detectors` on the output reduced workspace and check the relevant detectors are in the expected groups.

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
